Modules/_sha3/sha3module.c

```diff
@@ -272,9 +272,8 @@ SHA3_dealloc(SHA3object *self)
     if (self->lock) {
         PyThread_free_lock(self->lock);
     }
-    PyObject_Del(self);
-
     Py_DECREF(Py_TYPE(self));
```

Comment:
I would prefer to DECREF the type after calling PyObject_Del, see Modules/_abc.c for example. Something like:

```
    PyTypeObject *tp = Py_TYPE(self);
    (...)
    tp->tp_free(self);
    Py_DECREF(tp);
```

Reply:
@vstinner Got it, I will try doing it this way.